Add CONTRIBUTING.md file

hastagAB opened the pull request:
Added CONTRIBUTING.md file 
Issue Ref: #6 

Took Reference from:
1. https://github.com/emdupre/me-ica/blob/py3/CONTRIBUTING.md
2. https://github.com/KirstieJane/STEMMRoleModels/blob/gh-pages/CONTRIBUTING.md

Please suggest any changes or add-ons if required.

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+We're excited that you're here and equally thrilled to know that you want to contribute. :superhero:
+
+BIDS-Apps is the collection of containerized neuroimaging workflows and pipelines that accept datasets organized according to the Brain Imaging Data Structure (BIDS). To learn more please visit the [BIDS-Apps Webpage](http://bids-apps.neuroimaging.io)
```
```suggestion
BIDS-Apps are a collection of containerized neuroimaging workflows and pipelines that accept datasets organized according to the Brain Imaging Data Structure (BIDS). To learn more please visit our [Homepage](http://bids-apps.neuroimaging.io)
```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+## 3. Create a pull request :zap: 
+
+Pull requests with patches, improvements and new features are a great help. Please keep them clean from unwanted commits (One Commit One PR Rule)
```
perhaps linking to this best practice to help teach why this may be a good approach?

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+cd bids-apps.github.io
+
+# Assign the original repo to a remote called "upstream"
+git remote add upstream https://github.com/BIDS-Apps/bids-apps.github.io.git
```
```suggestion
git remote add upstream git@github.com:BIDS-Apps/bids-apps.github.io.git
```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+[Open a Pull Request.](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+
+A maintainer/developer will review and might suggest some changes before merging them into the codebase.
```
```suggestion
A maintainer/developer will review and might suggest some changes before merging them into the repository.
```

franklin-feingold commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,132 @@
+git checkout -b <topic-branch-name> 
+```
+
+#### 4. Locally rebase the upstream master branch into your topic branch:
````
perhaps rebasing for each contribution may set the barrier to contribute to high. This may be taken in the context of how a contribution to this repository would look like

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+## 5. A Special Note :high_brightness:
+
+Thank you for your valuable contribution! :smiley: 
```
```suggestion
Thank you for reading through and we look forward to your valuable contribution! :smiley: 
```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+We appreciate the hard work and are so grateful of the contributors who have added core features and improved existing ones throughout the history of the project. :raised_hands:
+
+Keep up the good work, the community remembers you. :bowtie: 
```
```suggestion

```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+Thank you for your valuable contribution! :smiley: 
+
+We appreciate the hard work and are so grateful of the contributors who have added core features and improved existing ones throughout the history of the project. :raised_hands:
```
```suggestion
We appreciate the hard work and time of our contributors who have built and maintained BIDS-Apps! :raised_hands:
```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+## 4. Get in touch with us :mailbox_with_mail: 
+
+If you have any question, comment or suggestion, feel free to create an issue on our repo because that will be the fastest and easiest way of communicating us.
```
```suggestion
If you have any questions, comments, or suggestions, please feel free to create an issue [here](https://github.com/BIDS-Apps/bids-apps.github.io/issues)! A Maintainer or community member will follow up.
```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+- ![Question](https://img.shields.io/badge/-question-%23cc317c) *It indicates that an issue or pull request needs more information.*
+These issues are some questions regarding the project that might need clarifications and a bit of discussion. 
+
+## 3. Create a pull request :zap: 
```
preserve consistency between our [bids-specification](https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md) and [bids-starter kit](https://github.com/bids-standard/bids-starter-kit/blob/master/CONTRIBUTING.md) contributing guides

```suggestion
## 3. Making a change with a pull request :zap: 
```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+Pull requests with patches, improvements and new features are a great help. Please keep them clean from unwanted commits (One Commit One PR Rule)
+
+Follow the steps to get your work included in the project.
```
more stylistic - I think we want to nurture contributions rather than making them appear rigidly defined 

```suggestion

```

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+#### 5. Open a Pull Request with a clear title and description against the master branch.
+
+[Open a Pull Request.](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
```
perhaps this step can use our [existing documentation ](https://github.com/bids-standard/bids-starter-kit/blob/master/CONTRIBUTING.md#example-pull-request)

franklin-feingold commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,132 @@
+
+***
+
+## 5. A Special Note :high_brightness:
```
May also be beneficial to add a section on how we [recongize contributions in BIDS](https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md#recognizing-contributions)

franklin-feingold commented:
thank you for putting together this pull request @hastagAB ! 

Overall, this could be a good starting place for building out the Contributing Guide for BIDS-Apps! One general question I had after reading through was how a contribution to this repository would look like (e.g. adding an App). This could be a future contribution.

I've gone through and added some comments. I think it has elements of our [BIDS starter kit Contributing Guide](https://github.com/bids-standard/bids-starter-kit/blob/master/CONTRIBUTING.md) and [BIDS specification Contributing Guide](https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md) throughout. I think it would be beneficial to preserve a level of consistency across the repositories.

hastagAB commented:
Yes, thanks for the feedback. Now it looks more consistent and similar to other BIDS repositories.

I've added the label and squashed the commits. Please have a look.

hastagAB commented:
Thanks! Looking forward to contributing more to the BIDS Community! :smiley:

franklin-feingold commented:
thanks @hastagAB !
